pkg/controller/direct/dataform/repository_controller.go

```diff
@@ -191,16 +190,43 @@ func (a *Adapter) Update(ctx context.Context, u *unstructured.Unstructured) erro
 	updateMask := &fieldmaskpb.FieldMask{}
 
 	if a.desired.Spec.GitRemoteSettings != nil {
-		if !reflect.DeepEqual(a.desired.Spec.GitRemoteSettings, a.actual.GitRemoteSettings) {
+		mapCtx := &direct.MapContext{}
+		protoDesired := RepositoryGitRemoteSettings_ToProto(mapCtx, a.desired.Spec.GitRemoteSettings)
```

Comment:
I would expect that  DataformRepositorySpec_ToProto would have done this already (we might well need to override  DataformRepositorySpec_ToProto to make this call)

Reply:
Chatting offline -- I think we have to convert the individual pieces to proto since `a.desired` is KRM based here. I was following along the cbwp as it was the latest (at the time?). I see in monitoringdashboards both desired and actual are proto-based 🤔